charmhelpers/contrib/openstack/cert_utils.py

```diff
@@ -307,6 +307,26 @@ def install_certs(ssl_dir, certs, chain=None, user='root', group='root'):
             content=bundle['key'], perms=0o640)
 
 
+def get_cert_relation_ca_name(cert_relation_id=None):
+    """Determine CA certificate name as provided by relation.
+
+    The filename on disk depends on the name chosen for the application on the
+    providing end of the certificates relation.
+
+    :param cert_relation_id: (Optional) Relation id providing the certs
+    :type cert_relation_id: str
+    :returns: CA certificate filename without path nor extension
+    :rtype: str
+    """
+    if cert_relation_id is None:
+        for cert_relation_id in relation_ids('certificates'):
+            break
+        else:
+            return ''
```

Comment:
I would test this code path in `test_get_cert_relation_ca_name` as well.

Reply:
This also feels like a really awkward piece of code.  I think it is doing this:

```python
if cert_relation_id is None:
    try:
        cert_relation_id = relation_ids('certificates')[0]
    except IndexError:
        return ''
```

If it is, please consider changing it?